lib/xebow/rgb_matrix/animations/cycle_all.ex

```diff
@@ -4,33 +4,37 @@ defmodule Xebow.RGBMatrix.Animations.CycleAll do
   """
 
   alias Chameleon.HSV
-  alias Xebow.RGBMatrix
-  alias Xebow.RGBMatrix.Animation
+
+  # alias Xebow.RGBMatrix
+  # alias Xebow.RGBMatrix.Animation
 
   import Xebow.Utils, only: [mod: 2]
 
-  @behaviour Animation
+  @delay_ms 17
+
+  # @behaviour Animation
 
-  @impl true
-  @spec tick(tick :: RGBMatrix.tick()) :: map
-  def tick(tick) do
-    speed = 100
+  # @impl true
+  def init do
```

Comment:
I actually like your previous code better with `@behaviour Animation` defining the functions that are required to be implemented, as well as breaking out `tick` and `speed` if they're required properties that are used for every animation.

Another approach if you continue on this route could be to rename `init` to `config` (or at least in my mind, `init` implies `GenServer`). Then have all of the properties as optional with `Animation` (?) supplying defaults when the animation is run. Or if the callback function has its required properties documented, maybe letting it crash on a bad config is fine.